src/storage/mvcc/reader/point_getter.rs

```diff
@@ -151,15 +155,27 @@ impl<S: Snapshot> PointGetter<S> {
             }
         }
 
+        let mut can_be_cached_by_lock = true;
         match self.isolation_level {
             IsolationLevel::Si => {
                 // Check for locks that signal concurrent writes in Si.
-                self.load_and_check_lock(user_key)?;
+                self.load_and_check_lock(user_key, Some(&mut can_be_cached_by_lock))?;
             }
             IsolationLevel::Rc => {}
         }
 
-        self.load_data(user_key)
+        let mut use_near_seek = false;
+        if let Some(can_be_cached) = can_be_cached {
```

Comment:
Maybe we can use `if let Some(true) = can_be_cached {`?

Reply:
If we know that there is a lock after ts, maybe no need to pre-seek anymore?

Reply:
Addressed